test/src/main/java/org/apache/accumulo/test/functional/MasterBulkFailureIT.java

```diff
@@ -0,0 +1,136 @@
+    final String tableName = getUniqueNames(1)[0];
+    c.tableOperations().create(tableName);
+    c.tableOperations().setProperty(tableName, Property.TABLE_MAJC_RATIO.getKey(), "1");
+    c.instanceOperations().setProperty(Property.TSERV_BULK_MAX_TABLET_OVERLAP.getKey(), "2");
+    // splits
+    SortedSet<Text> splits = new TreeSet<>();
+    for (int i = 1; i < 50; i++) {
+      splits.add(new Text(Integer.toString(i)));
+    }
+    c.tableOperations().addSplits(tableName, splits);
+
+    // create a bulk load file
+    final FileSystem fs = getCluster().getFileSystem();
+    final Path basePath = getCluster().getTemporaryPath();
+    CachedConfiguration.setInstance(fs.getConf());
+
+    final Path base = new Path(basePath, "testBulkLoad" + tableName);
+    fs.delete(base, true);
+    fs.mkdirs(base);
+
+    Path bulkFailures = new Path(base, "failures");
+    Path files = new Path(base, "files");
+    fs.mkdirs(bulkFailures);
+    fs.mkdirs(files);
+    FileSKVWriter writer = FileOperations.getInstance().newWriterBuilder()
+        .forFile(files.toString() + "/bulk." + RFile.EXTENSION, fs, fs.getConf())
+        .withTableConfiguration(AccumuloConfiguration.getDefaultConfiguration()).build();
+    writer.startDefaultLocalityGroup();
+    for (int j = 0; j < 5; j++) {
+      writer.append(new Key(Integer.toString(j)), new Value(new byte[0]));
+    }
+    writer.close();
+
+    Path file = new Path(files, "bulk." + RFile.EXTENSION);
+
+    c.tableOperations().importDirectory(tableName, files.toString(), bulkFailures.toString(),
+        false);
+
+    RemoteIterator<LocatedFileStatus> status = fs.listFiles(bulkFailures, false);
+    Assert.assertTrue(status.hasNext());
+    Path newPath = status.next().getPath();
+    Assert.assertFalse(status.hasNext());
+
+    // verify nothing loaded
+    Scanner s = c.createScanner(tableName, new Authorizations());
+    s.setRange(new Range());
+    Iterator<Map.Entry<Key,Value>> iterator = s.iterator();
+    Assert.assertFalse(iterator.hasNext());
+
+    // move the file back to its original location, reset the property higher and try again
+    fs.rename(newPath, file);
+
+    c.instanceOperations().setProperty(Property.TSERV_BULK_MAX_TABLET_OVERLAP.getKey(), "4");
+
+    c.tableOperations().importDirectory(tableName, files.toString(), bulkFailures.toString(),
+        false);
+
+    status = fs.listFiles(bulkFailures, false);
+    Assert.assertFalse(status.hasNext());
+
+    // verify the keys were loaded
+    s = c.createScanner(tableName, new Authorizations());
+    s.setRange(new Range());
+    iterator = s.iterator();
+    Assert.assertTrue(iterator.hasNext());
+    iterator.next();
+    Assert.assertTrue(iterator.hasNext());
+    iterator.next();
+    Assert.assertTrue(iterator.hasNext());
+    iterator.next();
+    Assert.assertTrue(iterator.hasNext());
+    iterator.next();
+    Assert.assertTrue(iterator.hasNext());
+    iterator.next();
+    Assert.assertFalse(iterator.hasNext());
```

Comment:
This should be done in a loop, like you did above to create the file.

Reply:
Better yet, just use Guava's `Iterables.size(s)`